### PR TITLE
Fixes NPE for invalid Uri

### PR DESCRIPTION
**Pull request by felipecsl**

### CODE REVIEW

**Comment by felipecsl** [comment]
@cdeonier

**Comment by cdeonier** [comment]
:+1:

**Comment by elihart** [comment]
lgtm

File: deeplinkdispatch/src/main/java/com/airbnb/deeplinkdispatch/DeepLinkEntry.java [hunk 1]
```diff
@@ -104,7 +104,7 @@ private static String parsePath(DeepLinkUri parsedUri) {
 
   boolean matches(String inputUri) {
     DeepLinkUri deepLinkUri = DeepLinkUri.parse(inputUri);
```
Comment (elihart) [inline deeplinkdispatch/src/main/java/com/airbnb/deeplinkdispatch/DeepLinkEntry.java]: so it's completely valid that the `parse` method returns null? should it be annotated with nullable? As far as the source of that crash, should we care about why the input uri is not valid?

Comment (felipecsl) [inline deeplinkdispatch/src/main/java/com/airbnb/deeplinkdispatch/DeepLinkEntry.java]: Yeah, from the javadocs:
```
  /**
   * Returns a new {@code OkUrl} representing {@code url} if it is a well-formed HTTP or HTTPS URL,
   * or null if it isn't.
   */
```

That's just a copy of OkHttp's OkHttp's [HttpUrl](https://github.com/square/okhttp/blob/master/okhttp/src/main/java/com/squareup/okhttp/HttpUrl.java) with minor changes to allow any scheme, so I didn't want to change it too much

**Pull request merged**